### DISCUSSION TITLE
Input Data Extraction from different datasets

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#2222] [General]
Hello @njzjz,

Q1. I am trying to reproduce [MD17](http://www.sgdml.org/#datasets)  and [QM9](https://figshare.com/collections/Quantum_chemistry_structures_and_properties_of_134_kilo_molecules/978904) result. I want to know how you have extracted data to provide input to DeePMD-kit for producing results for [paper](https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.120.143001).
If you have used dpdata which format you have used? 

Q2.  Can DeePMD accept equilibrium molecules as input rather than Molecular dynamics trajectory data ?

**Comment by njzjz** [comment]
I am not the author. Ping @wanghan-iapcm @jameswind for comments.

**Comment by AnuragKr** [comment]
@wanghan-iapcm  @jameswind  Can you please help me with a question mentioned above?

**Reply by AnuragKr** [reply]
@wanghan-iapcm  Please suggest comments  --
1. How to extract input data from [MD17](http://www.sgdml.org/#datasets) and [QM9](https://figshare.com/collections/Quantum_chemistry_structures_and_properties_of_134_kilo_molecules/978904) dataset for DeePMD-kit training ?
2. Can DeePMD accept equilibrium molecules as input rather than Molecular dynamics trajectory data?

**Reply by wanghan-iapcm** [reply]
1. the deepmd-kit data format is explained in https://docs.deepmodeling.com/projects/deepmd/en/master/data/index.html . 
2. yes.

**Reply by O-Nay** [reply]
Hello, dear @AnuragKr. I'm quite new to molecular dynamics, can you please show detailed instructions on how to train QM9 on DeepMD and output the results?

**Reply by AnuragKr** [reply]
@O-Nay  Thanks for your question, I have done training with MD-17, not QM-9. 
Steps --
1. Download the Numpy format for every molecule from this [link](http://www.sgdml.org/#datasets)
2. Create a Data folder that will contain type. raw and type_map.raw , nopbc(for organic molecule) file and set.000 folder.
3. Manually filled in the details of the type.raw and type_map.raw as per molecules.
4. Wrote a script to read data from numpy file and create a coord.npy, force.npy and energy.npy file inside data/set.000 folder as per [DeePMD-data-format](https://docs.deepmodeling.com/projects/deepmd/en/master/data/system.html) Read the descriptor of the data format before writing a script -- [link](http://quantum-machine.org/datasets/) 
For QM-9 you can refer to this script and then modify it to create data in DeePMD format -- https://github.com/atomistic-machine-learning/SchNet/blob/master/scripts/load_qm9.py

**Reply by O-Nay** [reply]
@AnuragKr  Please tell me, could you provide data for ethanol? Preferably a folder with set.000 and type.raw. Look at the file structure and the script itself. Forgive me for my impudence, it's hard to figure it out yet)

**Reply by AnuragKr** [reply]
@O-Nay  I am sharing a [link](https://drive.google.com/file/d/1_nkMrTkFgMb2gQ-nYukVFc_q9iIjNqHu/view?usp=sharing) to a few molecules for which I have created data. Please let me know if you find any difficulty in downloading.
